### ISSUE TITLE
improve cross compile support

### ISSUE THREAD

**Issue by andre2007** [dlang/dub#1259]
dub already supports cross compiling e.g. from windows to raspberry pi with following dub.json

```
{
	"name": "test",
	"dflags-ldc": ["-mtriple=arm-linux-gnueabihf", "-gcc=arm-linux-gnueabihf-gcc"],
	"lflags-ldc": ["-LC:\\D\\ldc2-1.4.0-beta1-win32-msvc\\bin\\ldc-build-runtime.tmp\\lib\\"]
}
```

and following command
dub build --compiler=ldc2

There resulting binary works fine on raspberry pi. There is only 1 minor issue. The resulting binary has the windows file extension ".exe". I wonder whether an additional parameter "targetExtenstion" in the json makes sense?

**Comment by MartinNowak** [comment]
Cross compilation shouldn't really be sth. you hardcode in a package configuration.
Instead the `--arch=` flag should be usable to select the target (would be better named `--target`).
Things like extensions should be inferred by dub.

**Comment by andre2007** [comment]
As far as I --arch understand you can only set x86 or x86_64 which can be linux or windows. I do not understand how the operation system can be determined by this parameter.

What would be the best way to set the DFLAGS LFLAGS parameters for cross compilation? I have an idea but this would also require a dub enhancement. See #1265

**Comment by s-ludwig** [comment]
This is the current plan regarding cross-compilation: https://github.com/dlang/dub/issues/228

**Comment by andre2007** [comment]
I think I found a really nice solution. Also the executable file has the right file ending and no adaption of dub.json is needed.

In the bin folder of ldc I created a batch file with the name ldc2pi and following content

```
@ECHO OFF

if "%1"=="-version" (
  ldc2 -version
) else (
  SET DFLAGS=-mtriple=arm-linux-gnueabihf -gcc=arm-linux-gnueabihf-gcc -L-L%~dp0\ldc-build-runtime.tmp\lib\
  ldmd2 %*
) 
```
To build a dub package for raspberry pi the only thing to do is to execute following command:
dub build --compiler=ldc2pi

Dub works really great. I will close this issue.
